### PR TITLE
[rllib] Persistent advantage learning

### PR DESCRIPTION
**Pull request by victorsun123**
## What do these changes do?

Add persistent advantage learning to rllib dqn.

## Related issue number

None

### CODE REVIEW

**Comment by robertnishihara** [comment]
@victorsun123 @richardliaw is this still in progress or should it be closed?

**Comment by richardliaw** [comment]
This is in progress - I'll get it merged in the next few days.

File: python/ray/rllib/dqn/models.py [hunk 1]
```diff
@@ -135,14 +150,31 @@ def __init__(
...
+        #import ipdb; ipdb.set_trace();
         # compute RHS of bellman equation
         q_t_selected_target = (
             rew_t + config["gamma"] ** config["n_step"] * q_tp1_best_masked)
 
+
+        #adjust for persistent advatange learning 
+        if config["pal"]:
+            #import ipdb; ipdb.set_trace()
+            first_diff = tf.reduce_max(self.q_t) - q_t_selected
```
Comment (richardliaw) [inline python/ray/rllib/dqn/models.py]: so actually this might not be right.. Let `q` be the online net and `Q` be the target net.

looks like the first diff is `V(s_t) - Q(s_t, a_t)`; or `max_a Q(s_t, . ) - Q(s_t, a_t)`.

However, here implemented, `self.q_t` is `q` rather than `Q`; resulting in `first_diff = max_a q(s_t, . ) - q(s_t, a_t)`. 

Can you fix this?

Comment (richardliaw) [reply python/ray/rllib/dqn/models.py]: Would be great if you could show some graphs of before and after too (just paste them on this PR).

**Review by richardliaw — CHANGES_REQUESTED** [review]

**Comment by robertnishihara** [comment]
@richardliaw just checking up on this. Is this still in progress?

**Comment by richardliaw** [comment]
Closing this for now.
